docs/user/OIDC/OAuth2OIDC-keycloak.md

```diff
@@ -1,9 +1,310 @@
+
+The following command will generate the private key and certificate files:
+
+```shell
+openssl req \
+      -newkey rsa:2048 -nodes -keyout certificate.key \
+      -x509 -days 365 -config certificate.config -out certificate.crt
+```
+
+#### Step 2: create the keystore and truststore
+
+Creating the keystore is done in two steps, starting by creating a pkcs12 keystore and then importing it using keytool (this will require a password, that we will need in the helm values.yaml):
+
+```shell
+openssl pkcs12 -export -in certificate.crt -inkey certificate.key -name keycloak -out certificate.p12
+
+keytool -importkeystore -srckeystore certificate.p12  -srcstoretype pkcs12 -destkeystore keycloak-0.keystore.jks -alias keycloak
+```
+
+Then we need to create the truststore:
+```shell
+keytool -import -alias keycloak -file certificate.crt -keystore keycloak.truststore.jks
+```
+
+#### Step 3: create a k8s secret
+
+The last step is to create a secret in the namespace where Keycloak will be installed.
+
+Note that the names of the keystore and truststore matters and must be exactly as written here (make sure to rename them if the names have been changed in the commands in Step 2):
+
+```shell
+kubectl create secret generic keycloak-tls --from-file=./keycloak-0.keystore.jks  --from-file=./keycloak.truststore.jks
+```
+
+## Helm Install
+
+To provide a default install, not many values must be provided in values.yaml - this is mostly some default passwords and the name of the secret created in Step 3 above.
+
+Here is the values.yaml file that i used:
+
+```yaml
+## Keycloak authentication parameters
+## ref: https://github.com/bitnami/bitnami-docker-keycloak#admin-credentials
+##
+auth:
+
+ ## Create administrator user on boot.
+ ##
+ createAdminUser: true
+
+ ## Keycloak administrator user and password
+ ##
+ adminUser: admin
+ adminPassword: Vmware!23
+
+ ## Wildfly management user and password
+ ##
+ managementUser: manager
+ managementPassword: Vmware!23
+
+ ## TLS encryption parameters
+ ## ref: https://github.com/bitnami/bitnami-docker-keycloak#tls-encryption
+ ##
+ tls:
+   enabled: true
+
+   ## Name of the existing secret containing the truststore and one keystore per Keycloak replica
+   ##
+   jksSecret: keycloak-tls
+
+   ## Password to access the keystore when it's password-protected.
+   ##
+   keystorePassword: Vmware!23
+   ## Password to access the truststore when it's password-protected.
+   ##
+   truststorePassword: Vmware!23
+```
+
+Then just deploy Keycloak either using Kubeapps or helm cli.
+
+
+# Configuration
+
+Follow the Keycloak documentation to create and configure a new Realm to work with.
+
+This section will focus on a few aspects to configure for the SSO scenario to work.
+
+## Groups Claim
+
+By default, there is no “groups” scope/claim. We will create a global client scope for groups.
```

Comment:
```suggestion
By default, there is no "groups" scope/claim. We will create a global client scope for groups.
```

Reply:
This suggestion also applies to the rest of the right/left quotation marks`“` (unicode 201C and 201D) ---> `"` (unicode 0022)